### PR TITLE
Fix cython3

### LINKED ISSUE #34014: Update setting data pointers for Cython 3
Need to update the following locations

- [x] _libs/window/aggregations.pyx: `bufarr.data`
- [ ] _libs/reduction.pyx: `chunk.data`

---

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=34888&view=logs&j=33ccdd52-c922-5ef2-8209-78215e36d994&t=046ffbec-62c2-54e3-e88a-7745f4292bb6&l=457

just started failing.

cc @pandas-dev/pandas-core @pandas-dev/pandas-triage 

if anyone has insights

### PR DESCRIPTION
**Pull request by tacaswell**
I'm not sure if this actually works, but it gets cython to compile with py310.

- [x] closes #34014
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Should this take over from #34997?

**Comment by tacaswell** [comment]
Ah, I missed #34997.  I probably should pick off the second commit into its own PR and then sort out if my attempt here is the right path or the one in the other PR.

That said, my guess is that this code is performance critical and was written the way it was as an optimization and I really don't have more than a passing understanding of the block manager so I am not the right person (and don't really have the bandwidth right now to become the right person) to take this on...

File: pandas/_libs/writers.pyx [hunk 1]
```diff
@@ -144,7 +144,7 @@ cpdef inline Py_ssize_t word_len(object val):
         Py_ssize_t l = 0
 
     if isinstance(val, str):
-        l = PyUnicode_GET_SIZE(val)
+        l = PyUnicode_GET_LENGTH(val)
```
Comment (jbrockmendel) [inline pandas/_libs/writers.pyx:147]: this can be done independently of the rest of the PR?

Comment (tacaswell) [reply pandas/_libs/writers.pyx:147]: Yeah, do you want to split it out?

**Comment by tacaswell** [comment]
Sorry for both the low-quality PR and dropping the ball on moving this forward.  I had this sitting on my machine for a while,  thought it was better to get it out in the public at all and that I would have more bandwidth to follow up :(.

**Comment by WillAyd** [comment]
All good @tacaswell ! I think this is getting covered in some other PRs so closing this one to clear the queue, but effort much appreciated in any case

**Pull request closed without merging**